src/plugins/query_enhancements/public/datasets/s3_type.ts

```diff
@@ -246,3 +255,62 @@ const fetchTables = async (http: HttpSetup, path: DataStructure[]): Promise<Data
 
   return fetch(http, path, 'TABLE');
 };
+
+// Function to process the input and map types using the new SQL to OSD mapping
+export function mapResponseToFields(sqlOutput: {
+  status: string;
```

Comment:
Yes this would be common for PPL/SQL and async queries. Converted to a type [here](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/8337/files#diff-e3aee9041606e7aa03e0c86eb518a3a15d1ce3a5d2af35f1064be36b417270ffR38)